File: 1519-minimum-subsequence-in-non-increasing-order/1519-minimum-subsequence-in-non-increasing-order.py
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums.sort(reverse=True)

        totalSum = 0
        for num in nums:
            totalSum += num
        
        Subsequence = []
        SubSum = 0

        for num in nums:
            SubSum += num
            Subsequence.append(num)
            if SubSum > totalSum - SubSum:
                break

        return Subsequence
        

File: 1519-minimum-subsequence-in-non-increasing-order/1519-minimum-subsequence-in-non-increasing-order.java
class Solution {
    public List<Integer> minSubsequence(int[] nums) {
        Arrays.sort(nums);

        //function to reversed the array
        reverseArr(nums);

        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }

        int subSum = 0;
        List<Integer> subsequence = new ArrayList<>();

        for (int num : nums) {
            subSum += num;
            subsequence.add(num);

            if (subSum > totalSum - subSum) {
                break;
            }
        }

        return subsequence;
    }

    public static void reverseArr(int[] arrays){
        int left = 0;
        int right = arrays.length - 1;
        while (left < right) {
            int temp = arrays[left];
            arrays[left] = arrays[right];
            arrays[right] = temp;
            left++;
            right--;
        }
    }
}